setup.py

```diff
@@ -45,7 +45,9 @@
 if setup_configure.mpi_enabled():
     RUN_REQUIRES.append('mpi4py >=3.0.2')
     SETUP_REQUIRES.append("mpi4py ==3.0.2; python_version<'3.8'")
-    SETUP_REQUIRES.append("mpi4py ==3.0.3; python_version>='3.8'")
+    SETUP_REQUIRES.append("mpi4py ==3.0.3; python_version=='3.8.*'")
+    SETUP_REQUIRES.append("mpi4py ==3.0.3; python_version=='3.9.*'")
+    SETUP_REQUIRES.append("mpi4py ==3.1.3; python_version>='3.10'")
```

Comment:
This is the latest version version of mpi4py, is it really the minimum version we can use?

Reply:
https://pypi.org/project/mpi4py/3.1.0/ 
Official support for python 3.10 (and 3.9 actually) was introduced with mpi4py 3.1.0
Python 3.9 is not listed as supported for version 3.0.3

@julesghub

Reply:
```
    SETUP_REQUIRES.append("mpi4py ==3.0.3; python_version>='3.8'")
    SETUP_REQUIRES.append("mpi4py ==3.1.0; python_version>='3.9'")
```
should be enough.

Reply:
Ok, I've set  it to use mpi4py 3.1.0 for py39 and py310.